src/titanium-search-input.ts

```diff
@@ -0,0 +1,152 @@
+import '../lib/titanium-svg-button';
+import {css, customElement, html, LitElement, property} from 'lit-element';
+
+@customElement('titanium-search-input')
+export class TitaniumSearchInput extends LitElement {
+  @property() value: string = '';
+  @property() placeholder: string = '';
+  @property({type: Boolean, reflect: true}) disabled: boolean;
+
+  _input: HTMLElement|any;
+
+  firstUpdated() {
+    super.firstUpdated(new Map());
+    const el = this.shadowRoot && this.shadowRoot.querySelector('input');
+    if (el)
+      this._input = el;
+  }
+
+  _onValueChange() {
+    this.value = this._input.value;
+    this.dispatchEvent(new CustomEvent('value-changed', {detail: this.value}));
+  }
+
+  _onClearClick() {
+    if (this.disabled)
+      return;
+    this.value = '';
+  }
+
+  focus() {
+    this._input.focus();
+  }
+
+  static styles = css`
+  :host {
+    display: block;
+  }
+
+  input-container {
+    @apply --layout-self-end;
```

Comment:
Lets not use iron-layout in these root components,

Reply:
I will address this.